pkg/api/controllers/fileshare_test.go

```diff
@@ -368,3 +536,230 @@ func TestUpdateFileShareSnapshot(t *testing.T) {
 		assertTestResult(t, w.Code, 500)
 	})
 }
+
+func TestDeleteFileShareSnapshot(t *testing.T) {
+
+	t.Run("Should return 202 if everything works well", func(t *testing.T) {
+		//var snapshot []*model.FileShareSnapshotSpec
+		mockClient := new(dbtest.Client)
+		mockClient.On("GetFileShareSnapshot", c.NewAdminContext(), "3769855c-a102-11e7-b772-17b880d2f537").Return(&SampleFileShareSnapshots[0], nil)
+		mockClient.On("GetProfile", c.NewAdminContext(), SampleFileShareSnapshots[0].ProfileId).Return(&SampleFileShareProfiles[0], nil)
+		mockClient.On("GetFileShare", c.NewAdminContext(), SampleFileShareSnapshots[0].FileShareId).Return(&SampleFileShares[0], nil)
+		mockClient.On("UpdateFileShareSnapshot", c.NewAdminContext(), SampleFileShareSnapshots[0].Id, &SampleFileShareSnapshots[0]).Return(nil, nil)
+		mockClient.On("DeleteFileShareSnapshot", c.NewAdminContext(), "3769855c-a102-11e7-b772-17b880d2f537").Return(nil)
+		db.C = mockClient
+
+		r, _ := http.NewRequest("DELETE",
+			"/v1beta/file/snapshots/3769855c-a102-11e7-b772-17b880d2f537", nil)
+		w := httptest.NewRecorder()
+		beego.InsertFilter("*", beego.BeforeExec, func(httpCtx *context.Context) {
+			httpCtx.Input.SetData("context", c.NewAdminContext())
+		})
+		beego.BeeApp.Handlers.ServeHTTP(w, r)
+		assertTestResult(t, w.Code, 202)
+	})
+
+	t.Run("Should return 500 if delete file share snapshot with bad request", func(t *testing.T) {
```

Comment:
Why Bad request is 500 error? Should it not be 400?

Reply:
I will check once, may be we need to change main code

Reply:
Is this resolved?